external-crates/move/crates/move-ir-compiler/src/unit_tests/testutils.rs

```diff
@@ -19,7 +20,8 @@ fn compile_module_string_impl(
 
     let mut serialized_module = Vec::<u8>::new();
     compiled_module.serialize_with_version(compiled_module.version, &mut serialized_module)?;
-    let deserialized_module = CompiledModule::deserialize_with_defaults(&serialized_module)
+    let config = BinaryConfig::with_extraneous_bytes_check(false);
```

Comment:
Not sure why we're setting `with_extraneous_bytes_check` to `false` here. Should probably be `true`.

Reply:
In both cases, I was just matching existing behavior: `deserialzie_with_defaults` sets it to `false`, and was previously used everywhere. I have changed that to set it to `true`, and updated these cases.